mbuild/tests/test_rigid.py

```diff
@@ -0,0 +1,365 @@
+import pytest
+import numpy as np
+
+import mbuild as mb
+from mbuild.tests.base_test import BaseTest
+from mbuild.utils.io import get_fn
+
+
+class TestRigid(BaseTest):
+
+    def test_load_rigid(self, rigid_benzene):
+        assert rigid_benzene.rigid is True
+        assert rigid_benzene.rigid_id is 0
+        assert rigid_benzene.max_rigid() is 0
+        assert len([p for p in rigid_benzene.rigid_particles()]) == 12
+        assert [p for p in rigid_benzene.rigid_ids()].count(0) == 12
+
+    def test_load_nonrigid(self, benzene):
+        assert benzene.rigid is False
+        assert benzene.rigid_id is None
+        assert benzene.max_rigid() is None
+        assert len([p for p in benzene.rigid_particles()]) == 0
+        assert [p for p in benzene.rigid_ids()].count(None) == 12
+
+    def test_rigid_from_parts(self, rigid_ch):
+        benzene = mb.Compound()
+        benzene.add(rigid_ch)
+        current = rigid_ch
+
+        for _ in range(5):
+            ch_new = mb.clone(rigid_ch)
+            mb.force_overlap(move_this=ch_new,
+                             from_positions=ch_new['a'],
+                             to_positions=current['b'])
+            current = ch_new
+            benzene.add(ch_new, increment_rigid=False)
+
+        carbons = [p for p in benzene.particles_by_name('C')]
+        benzene.add_bond((carbons[0],carbons[-1]))
+
+        assert benzene.rigid is True
+        assert benzene.rigid_id is None
+        assert benzene.max_rigid() is 0
+        assert len([p for p in benzene.rigid_particles()]) == 12
+        assert [p for p in benzene.rigid_ids()].count(0) == 12
+
+    def test_rigid_from_parts2(self, rigid_ch):
+        benzene = mb.Compound(rigid=True)
+        benzene.add(rigid_ch, increment_rigid=False)
+        current = rigid_ch
+
+        for _ in range(5):
+            ch_new = mb.clone(rigid_ch)
+            mb.force_overlap(move_this=ch_new,
+                             from_positions=ch_new['a'],
+                             to_positions=current['b'])
+            current = ch_new
+            benzene.add(ch_new, increment_rigid=False)
+
+        carbons = [p for p in benzene.particles_by_name('C')]
+        benzene.add_bond((carbons[0],carbons[-1]))
+
+        assert benzene.rigid is True
+        assert benzene.rigid_id is 0
+        assert benzene.max_rigid() is 0
+        assert len([p for p in benzene.rigid_particles()]) == 12
+        assert [p for p in benzene.rigid_ids()].count(0) == 12
+
+    def test_nonrigid_from_parts(self, benzene_from_parts):
+        assert benzene_from_parts.rigid is False
+        assert benzene_from_parts.rigid_id is None
+        assert benzene_from_parts.max_rigid() is None
+        assert len([p for p in benzene_from_parts.rigid_particles()]) == 0
```

Comment:
You can just write `len(list(foo.rigid_particles()))`. You've got a few of them sprinkled around